### PR TITLE
2046 transfer must use token validation package

### PR DESCRIPTION
**Pull request by sahma19**

### CODE REVIEW

File: domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs [hunk 1]
```diff
@@ -238,7 +248,7 @@ private static async Task InsertTransferAgreementHistoryEntry(ApplicationDbConte
         var historyQuery =
             $"INSERT INTO \"{historyTable}\" (\"Id\", \"CreatedAt\", \"AuditAction\", \"ActorId\", \"ActorName\", \"TransferAgreementId\", \"StartDate\", \"EndDate\", \"SenderId\", \"SenderName\", \"SenderTin\", \"ReceiverTin\") " +
             "VALUES (@Id, @CreatedAt, @AuditAction, @ActorId, @ActorName, @TransferAgreementId, @StartDate, @EndDate, @SenderId, @SenderName, @SenderTin, @ReceiverTin)";
-        var historyFields = new[]
+        object[] historyFields =
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs:251]: And this?

File: domains/transfer/Transfer.API/API/appsettings.Development.json [hunk 2]
```diff
@@ -25,5 +25,8 @@
         "BaseUrl": "http://distribution.virk.dk",
         "User": "SomeUser",
         "Password": "SomePassword"
-    }
+    },
+    "TokenValidation:Issuer": "Us",
+    "TokenValidation:Audience": "Users",
```
Comment (tnickelsen) [inline domains/transfer/Transfer.API/API/appsettings.Development.json:30]: Is this good enough?

File: domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs [hunk 3]
```diff
@@ -215,7 +225,7 @@ private static async Task InsertTransferAgreement(ApplicationDbContext dbContext
 
         var agreementQuery =
             $"INSERT INTO \"{agreementsTable}\" (\"Id\", \"StartDate\", \"EndDate\", \"SenderId\", \"SenderName\", \"SenderTin\", \"ReceiverTin\", \"ReceiverReference\", \"TransferAgreementNumber\") VALUES (@Id, @StartDate, @EndDate, @SenderId, @SenderName, @SenderTin, @ReceiverTin, @ReceiverReference, @TransferAgreementNumber)";
-        var agreementFields = new[]
+        object[] agreementFields =
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs:228]: Dotnet 8, introduces new warnings, that are resolved, by doing it like this.

File: domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs (hunk 1 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Factories/TransferAgreementsApiWebApplicationFactory.cs:251]: Dotnet 8, introduces new warnings, that are resolved, by doing it like this.

File: domains/transfer/Transfer.API/API/appsettings.Development.json (hunk 2 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/appsettings.Development.json:30]: Yes. This token is only for local development. ENV_Variables, in eo-base overwrite these when deployed to prod

File: domains/transfer/Transfer.API/API/Program.cs [hunk 4]
```diff
@@ -87,19 +86,11 @@
...
-            ValidateAudience = false,
-            // Validate life time disabled as the JWT token generated from the auth service wrongly names the claim for expiration
-            ValidateLifetime = false,
-            SignatureValidator = (token, _) => new JwtSecurityToken(token)
-        };
-    });
+var tokenValidationOptions = builder.Configuration.GetSection(TokenValidationOptions.Prefix).Get<TokenValidationOptions>()!;
+builder.Services.AddOptions<TokenValidationOptions>().BindConfiguration(TokenValidationOptions.Prefix).ValidateDataAnnotations().ValidateOnStart();
+
+builder.AddTokenValidation(tokenValidationOptions);
```
[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Program.cs:92]: What do you mean?

[the remark this answers is not in the document]
Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/Program.cs:92]: I mean the TokenValidationParameters you deleted (I should have marked more code)

[the remark this answers is not in the document]
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Program.cs:92]: They are no longer needed. The TokenValidationPackage has been refactored. The Method only requires TokenValidationOptions now, and generates parameters, automatically from them

**Review by tnickelsen — APPROVED** [review]

**Pull request merged**